src/services/component.browser.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { flushPromises, mount } from '@vue/test-utils'
+import { expect, test } from 'vitest'
+import { defineAsyncComponent } from 'vue'
+import echo from '@/components/echo'
+import { component } from '@/services/component'
+import { createRouter } from '@/services/createRouter'
+import { createRoutes } from '@/services/createRoutes'
+
+test('renders component with props', async () => {
+  const routes = createRoutes([
+    {
+      name: 'one',
+      path: '/one',
+      component: component(echo, () => ({ value: 'one' })),
+    },
+    {
+      name: 'two',
+      path: '/two',
+      component: component(echo, async () => {
+        return await { value: 'two' }
+      }),
+    },
+    {
+      name: 'three',
+      path: '/three',
+      component: component(defineAsyncComponent(() => import('@/components/echo')), () => ({ value: 'three' })),
```

Comment:
Something interesting to note is I couldn't get this to work with just the dynamic import. I had to wrap it in `defineAsyncComponent`. The types work just fine, but I couldn't get the component to actually render.